### PR TITLE
Update to GSON 2.8.6

### LINKED ISSUE #488: Follow Orbits GSON Update
https://bugs.eclipse.org/bugs/show_bug.cgi?id=569966

### PR DESCRIPTION
**Pull request by apupier**
Fixes #488

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: releng/releng-target/lsp4j.target.target [hunk 1]
```diff
@@ -3,9 +3,9 @@
 <target name="org.eclipse.xtext.helios.target" sequenceNumber="0">
 	<locations>
 		<location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="false" type="InstallableUnit">
-			<repository location="https://download.eclipse.org/tools/orbit/downloads/2019-12"/>
-			<unit id="com.google.gson" version="2.8.2.v20180104-1110"/>
-			<unit id="com.google.gson.source" version="2.8.2.v20180104-1110"/>
+			<repository location="https://download.eclipse.org/tools/orbit/downloads/2021-03"/>
```
Comment (apupier) [inline releng/releng-target/lsp4j.target.target:6]: as 2021-03 not yet available, i guess we need to use https://download.eclipse.org/tools/orbit/downloads/drops/I20201231185508/repository for now.

**Comment by cdietrich** [comment]
we still need to wait for M1 of orbit the be published later this week

**Comment by apupier** [comment]
> we still need to wait for M1 of orbit the be published later this week

M1 has been published https://download.eclipse.org/tools/orbit/downloads/drops/S20210105214148/

**Comment by cdietrich** [comment]
Yes build is green now. PR waits for review

File: releng/releng-target/lsp4j.target.target (hunk 1 above)
Comment (apupier) [reply releng/releng-target/lsp4j.target.target:6]: 2021-03 M1 published, no need to change

**Comment by cdietrich** [comment]
can anyone please review

**Review by jonahgraham — APPROVED** [review]
> can anyone please review

Sorry - missed your review request. Thanks for the ping.

I don't see any more or less issue here than last time GSON upgraded (#310) , but some of GSON types are part of lsp4j API and as such this is an API breaking change and should be documented if so.  That documentation wasn't done for #310, so perhaps my recollection is wrong.

**Comment by cdietrich** [comment]
where do you document such things?

**Comment by jonahgraham** [comment]
PS @cdietrich If there is anything specific you were hoping to have reviewed about this please let me know. I had seen your initial reply to @apupier so I assumed you were the one doing the review and stopped paying attention.

**Comment by jonahgraham** [comment]
> where do you document such things?

In the changelog. See "Breaking API changes:" section for each release.

**Comment by cdietrich** [comment]
as we state gson 2.8.x in feature xml we seem to consider such a bugfix change as compatible

**Pull request merged**